### DISCUSSION TITLE
After Z2M crash all paired devices stuck in last state

### DISCUSSION THREAD

**Discussion by vyruz1986** [Koenkk/zigbee2mqtt#16634] [Devices]
I start today to migrate all my devices over from deconz/raspbee2 to Z2M/zigstar POE, i decided to just setup a new network and repair all devices to Z2M since that seemed the least error prone approach. after I paired about 15 devices to the new Z2M instance, something happened and my Z2M instance (running as home assistant addon) stopped, I didn't pay much attention and just restarted it and continued pairing devices. After a while I realised that all 15 devices I paired before this 'crash' happened, were not working anymore. Strange thing is they still show up in Z2M with signal strength and state (e.g. motion sensor reporting occupance, while there is no motion anymore). Giving any command (e.g. turning on light) causes an error message 'MAC no Ack' after a timeout period.
If I take such a device and re-pair it, the issue seems solved, but I would like to get to the bottom of what is wrong with those devices and how i can prevent it from happening again, since I'm not looking forward to re-pairing all my devices every week...

**Comment by klada** [comment]
Zigbee2mqtt normally dumps a backup of the coordinator state as a JSON file to the Z2M data directory. This is done on clean shutdown. So whenever the coordinator has an invalid state or something else happens, the backup should help you out there.

Now since it sounds like this was your first Z2M startup of a new instance and this has crashed, my guess is that there is no such backup (or an incomplete one). Normally the coordinator also has it's own state in NVRAM so this should not be too bad. But if this state is also messed up for whatever reason I think you'll end up with the behavior you are describing.

So long story short: after pairing your devices you should restart zigbee2mqtt cleanly so it creates this backup file. This will then help you out in case something goes wrong later.

**Comment by vyruz1986** [comment]
Thanks for clarifying @klada !
I think I have some other issue though, I started this morning repairing the devices which were not working after yesterday's 'crash'. But after a couple of minutes I started again having similar issues, this time withouth Z2M having crashed. 

I use Z2M at another location with a Sonoff USB dongle, and it works perfectly.
But at my home, since my Z2M server is located far away from most zigbee devices, I decided to the [Zigstar Olizig POE gateway](https://zig-star.com/projects/zigstar-olizig/) a try.
It uses the TCP serial connection from Z2M to connect to the gateway.

Could it be that this serial over network connection is just too unstable?

Sidenote: I did try to use ZHA with this gateway, but there I get the following error: `Network formation refused: there is too much RF interference. Make sure your coordinator is on a USB 2.0 extension cable and away from any sources of interference, like USB 3.0 ports, SSDs, 2.4GHz routers, motherboards, etc` The Zigstar GW is sitting right next to where my previous Raspbee2/deconz gateway was located, and I never had issues with my zigbee network there...

**Reply by klada** [reply]
> Could it be that this serial over network connection is just too unstable?

It depends on your network. The network should be wired (no wireless bridges in between, just plain Ethernet) and you should generally not have any packet loss. 

I am using a POE gateway as well for the same reason as you and it is rock solid for me. The PoE-Gateway (I am using the tubeszb-cc2652-poe)  has been the best ZigBee investment ever for me :smile: 

> Sidenote: I did try to use ZHA with this gateway, but there I get the following error: Network formation refused: there is too much RF interference. Make sure your coordinator is on a USB 2.0 extension cable and away from any sources of interference, like USB 3.0 ports, SSDs, 2.4GHz routers, motherboards, etc

Have you tried a different ZigBee channel yet? The ZigBee channel should be as far away from your primary WiFi channel or right between your WiFi channes if you are using multiple APs. Have a look at https://www.metageek.com/training/resources/zigbee-wifi-coexistence/

To check the quality of your currently selected ZigBee channel you could also do an [energy scan](https://github.com/zigpy/zigpy-znp/blob/dev/TOOLS.md#energy-scan) with zigpy. Note that zigpy does not have anything to do with zigbee2mqtt, it's a completely different software/tool stack. But I found it quite helpful for planning my ZigBee network beforehand.

When doing an energy scan I'd recommend turning of all of the ZigBee devices that you are planning to use with the coordinator (if possible), as their traffic should not be counted during the scan.
